pkg/certificate/providers/tresor/debugger_test.go

```diff
@@ -1,30 +1,27 @@
 package tresor
 
 import (
-	"time"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
 	"github.com/openservicemesh/osm/pkg/certificate"
-	"github.com/openservicemesh/osm/pkg/certificate/pem"
 )
 
 var _ = Describe("Test Tresor Debugger", func() {
 	Context("test ListIssuedCertificates()", func() {
-		cert := &Certificate{
-			privateKey: pem.PrivateKey("yy"),
-			certChain:  pem.Certificate("xx"),
-			expiration: time.Now(),
-			commonName: "foo.bar.co.uk",
-		}
+		// Setup:
+		//   1. Create a new (fake) certificate
+		//   2. Reuse the same certificate as the Issuing CA
+		//   3. Populate the CertManager's cache w/ cert
+		cert := NewFakeCertificate()
 		cert.issuingCA = cert.GetCertificateChain()
```

Comment:
Correct me if wrong, but you are creating a fake speficic to later call interface on it. If that's the case, we need a mock of the interface instead of a fake.

The problem of the fakes is that, as the tests grow in complexity, they hide specifics inside that are later hard to change for different tests (see fake catalog, that didn't go well)

Reply:
From what I see a new helper is being added to initialize the certificate interface. I'd prefer to switch to mocks for the very reason that you mentioned regarding complexity growing, but this change looks small and isolated.

Reply:
fair